### PR TITLE
iox-#743 Rework and rename conceptual guide

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] ~~Code follows the coding style of [CONTRIBUTING.md][contributing]~~
1. [x] ~~Tests follow the [best practice for testing][testing]~~
1. [x] ~~Changelog updated [in the unreleased section][changelog] including API breaking changes~~
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer
* Remove duplicate information
* Rename `conceptual-guide.md` to `shared-memory-communication.md`

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #743 (partly)

### CODE REVIEW

File: doc/shared-memory-communication.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<!-- @todo Move the content of this file to website/getting-started/overview.md or website/for-developers/* -->
```
Comment (mossmaurice) [inline doc/shared-memory-communication.md:1]: I used `git mv doc/conceptual-guide.md doc/shared-memory-communication.md` to rename the file. However, GitHub does not detect it as rename. Please use your favourite diff tool when reviewing.

File: doc/shared-memory-communication.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+# Shared memory communication
```
Comment (mossmaurice) [inline doc/shared-memory-communication.md:3]: This is not a new file. I used `git mv doc/conceptual-guide.md doc/shared-memory-communication.md` to rename the file. However, GitHub does not detect it as rename. Please use your favourite diff tool when reviewing.

File: doc/shared-memory-communication.md (hunk 1 above)
Comment (mossmaurice) [inline doc/shared-memory-communication.md:1]: If time before `v2.0` this will be done on a separate PR.

**Review by elfenpiff — DISMISSED** [review]

File: doc/shared-memory-communication.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+A consumer must explicitly indicate when it has finished processing a particular memory chunk it has received.
+Memory chunks are returned to the pool once all attached consumers indicate they have finished.
+
+### A Note on pointers
+
+As already discussed, shared memory segments may be mapped to different memory areas in the virtual address space of a
+process.
+To deal with this, iceoryx utilizes specialized pointer types: the `iox::rp::RelativePointer` and
+the `iox::rp::RelocatablePointer`.
```
Comment (FerdinandSpitzschnueffler) [inline doc/shared-memory-communication.md:80]: Actually it's `iox::rp::relocatable_ptr`, but this will be changed sometime in the future.

**Review by FerdinandSpitzschnueffler — COMMENTED** [review]
I think there are some links to the conceptional guide in other files, e.g. overview.md. Please update them.
Since you removed some sections: Please also check if these links are still valid.

File: doc/shared-memory-communication.md (hunk 3 above)
Comment (mossmaurice) [reply doc/shared-memory-communication.md:80]: Good catch, thanks!

File: doc/website/advanced/configuration-guide.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ With that change, the footprint of the management segment is reduced to ~52.7 MB
 ## :material-memory: Configuring Mempools for RouDi
 
 RouDi supports several shared memory segments with different access rights, to limit the read and write access between different applications. Inside of these segments reside mempools where the user payload data for transfer is stored.
-Based on the [conceptual guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/conceptual-guide.md) the end-user may want to configure the mempools with the number of chunks and their size.
+Based on the [shared memory concept article](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/shared-memory-communication.md) the end-user may want to configure the mempools with the number of chunks and their size.
```
Comment (FerdinandSpitzschnueffler) [inline doc/website/advanced/configuration-guide.md:34]: Actually I don't understand why we refer to the article.

File: README.md [hunk 5]
```diff
@@ -88,8 +88,7 @@ Please see the dedicated [README.md](tools/docker/README.md) for information on
 
 * [Getting Started](doc/website/getting-started/overview.md)
 * [Installation Guide](doc/website/getting-started/installation.md)
-* [Concepts](doc/conceptual-guide.md)
```
Comment (FerdinandSpitzschnueffler) [inline README.md:91]: Would it make sense to link to `/doc/website/concepts/`?

Comment (mossmaurice) [reply README.md:91]: I don't think detailed background on how shared memory works is needed for users. The important article is `overview.md` and is linked two lines above.

File: doc/website/advanced/configuration-guide.md (hunk 4 above)
Comment (mossmaurice) [reply doc/website/advanced/configuration-guide.md:34]: Hmm :thinking: I thought of it as getting more background info of now. Thinking again, I guess you're right. Removed.

File: doc/website/getting-started/overview.md [hunk 6]
```diff
@@ -147,7 +148,7 @@ Shared memory is physical memory that is made accessible to multiple processes v
 virtual address spaces.
 
 For further information have a look at our
-[conceptual guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/conceptual-guide.md).
+[shared memory concept article](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/shared-memory-communication.md).
```
Comment (MatthiasKillat) [inline doc/website/getting-started/overview.md:151]: broken link

File: doc/shared-memory-communication.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+## The basics
+
+When a process in a POSIX system starts it is given its own virtual address space.
+
+The range that a virtual address space spans may be the same for different processes, however the data that is
+accessible at a particular address may be different for each process.
+
+A pointer in an application uses the virtual address space of the process it is running in.
+
+Within the virtual address space of a process there are a number of "memory areas" to where data is loaded or mapped.
```
Comment (MatthiasKillat) [inline doc/shared-memory-communication.md:14]:
```suggestion
Within the virtual address space of a process there are a number of "memory areas" where data is loaded or mapped to.
```

File: doc/shared-memory-communication.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+The number of segments used by an iceoryx system, along with the configuration of the mempools they contain, are
+provided to the system via configuration.
+
+The configuration can be provided at compile time (as a header) or at runtime (as a toml-formatted text file).
+See the [configuration guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/configuration-guide.md#configuring-mempools-for-roudi) for more details.
+
+## Zero-copy communication
+
+`popo::Publisher`s / `popo::Subscriber`s and `popo::Server`s / `popo::Client`s which are wired together can communicate
```
Comment (MatthiasKillat) [inline doc/shared-memory-communication.md:55]:
```suggestion
`popo::Publisher`s / `popo::Subscriber`s and `popo::Server`s / `popo::Client`s which are connected can communicate
```

File: doc/shared-memory-communication.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+via shared memory resulting in zero-copy communication. In this section `popo::Publisher` and `popo::Server` will be
+referred to as producers and `popo::Subscriber` and `popo::Client` as consumers.
+
+A producer has an assigned shared memory segment to which it may write its data to.
```
Comment (MatthiasKillat) [inline doc/shared-memory-communication.md:59]:
```suggestion
A producer has an assigned shared memory segment to which it may write its data.
```

File: doc/shared-memory-communication.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+In a POSIX system, this is decided purely based on file access permissions as memory segments are represented as
+virtual files.
+
+To output data, a producer reserves a memory chunk in its assigned memory segment.
+The iceoryx system will intelligently choose the smallest chunk size that can fit the output data structure.
```
Comment (MatthiasKillat) [inline doc/shared-memory-communication.md:64]:
```suggestion
The iceoryx system will choose the smallest chunk size that can fit the output data structure (best fit policy).
```

File: doc/website/getting-started/overview.md (hunk 6 above)
Comment (mossmaurice) [reply doc/website/getting-started/overview.md:151]: Once this PR is merged the link won't be broken anymore ;)

File: doc/shared-memory-communication.md (hunk 9 above)
Comment (mossmaurice) [reply doc/shared-memory-communication.md:59]: I'm not a native English speaker, but the current wording sounds better to me. As this guide was initially written by @ithier I would leave it as it is.

File: doc/shared-memory-communication.md (hunk 10 above)
Comment (mossmaurice) [reply doc/shared-memory-communication.md:64]: Left out the `(...)`.

**Review by MatthiasKillat — APPROVED** [review]

**Review by FerdinandSpitzschnueffler — APPROVED** [review]
